src/BIP44Node.ts

```diff
@@ -110,87 +92,51 @@ export class BIP44Node implements BIP44NodeInterface {
    * @param options.derivationPath - The rooted HD tree path that will be used
    * to derive the key of this node. Mutually exclusive with `key`.
    */
-  constructor({ depth, key, derivationPath }: BIP44NodeOptions) {
-    const _key = BIP44Node._parseKey(key);
-
+  static async create({
+    depth,
+    key,
+    derivationPath,
+  }: Omit<SLIP10NodeOptions, 'curve'>): Promise<BIP44Node> {
     if (derivationPath) {
-      if (_key) {
-        throw new Error(
-          'Invalid parameters: May not specify a derivation path if a key is specified. Initialize the node with just the parent key and its depth, then call BIP44Node.derive() with your desired path.',
-        );
-      }
-
-      if (depth) {
-        throw new Error(
-          'Invalid parameters: May not specify a depth if a derivation path is specified. The depth will be calculated from the path.',
-        );
-      }
-
-      if ((derivationPath as any).length === 0) {
-        throw new Error(
-          'Invalid derivation path: May not specify an empty derivation path.',
-        );
-      }
-
       const _depth = derivationPath.length - 1;
-      validateBIP44Depth(_depth);
-      this.depth = _depth as BIP44Depth;
 
+      validateBIP44Depth(_depth);
       validateBIP44DerivationPath(derivationPath, MIN_BIP_44_DEPTH);
-      this.keyBuffer = deriveKeyFromPath(derivationPath, undefined, this.depth);
-    } else if (_key) {
-      validateBIP44Depth(depth);
-      this.depth = depth as BIP44Depth;
+    }
 
-      this.keyBuffer = _key;
-    } else {
-      throw new Error(
-        'Invalid parameters: Must specify either key or derivation path.',
-      );
+    if (key) {
+      validateBIP44Depth(depth);
     }
 
-    Object.freeze(this);
+    const node = await SLIP10Node.create({
+      key,
+      derivationPath,
+      depth,
+      curve: secp256k1,
+    });
+
+    return new BIP44Node(node);
   }
 
-  /**
-   * Constructor helper for validating and parsing the `key` parameter. An error
-   * is thrown if validation fails.
-   *
-   * @param key - The key to parse.
-   * @returns A {@link Buffer}, or `undefined` if no key parameter was
-   * specified.
-   */
-  private static _parseKey(key: unknown): Buffer | undefined {
-    if (key === undefined || key === null) {
-      return undefined;
-    }
+  #node: SLIP10Node;
 
-    let bufferKey: Buffer;
-    if (Buffer.isBuffer(key)) {
-      if (!isValidBufferKey(key)) {
-        throw new Error(
-          'Invalid buffer key: Must be a 64-byte, non-empty Buffer.',
-        );
-      }
+  public get depth(): BIP44Depth {
+    validateBIP44Depth(this.#node.depth);
```

Comment:
To keep TypeScript happy without the need for a type cast.

Reply:
I think it's completely fine to cast here. Even though this validation function is extremely cheap, I think we should avoid using runtime validation in these situations as a matter of course. If we really want to avoid type casts we could genericize the depth in the `SLIP10Node` type, but that's probably best done as a follow-up.

Reply:
I think it's completely fine to cast here, because the validation we already perform convinces us that the internal node is valid.

Furthermore, although this validation function is extremely cheap, I think we should avoid using runtime validation in these situations as a matter of course. If we _really_ want to avoid type casts we could genericize the depth in the `SLIP10Node` type, but that would hardly be worth it, and probably best done as a follow-up.

```suggestion
    this.#node.depth as BIP44Depth;
```